File: Electronics_Shop_17.Services/Database/OrderItem.cs
﻿using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Electronics_Shop_17.Services.Database
{
    public class OrderItem : IHelperId
    {
        public int Id { get; set; }
        public int Quantity { get; set; }
        public double Price { get; set; }

        public int OrderId { get; set; }
        public int ProductId { get; set; }
        public virtual Product Product { get; set; }
        public double FinalPrice { get; set; }
        public int ProductColorId { get; set; }
        public virtual ProductColor ProductColor { get; set; }
    }
}

File: angular17_web/src/app/models/order-item/order.item.ts
import { Product } from '../product/product';
import { ProductColor } from '../product-color/product.color';

export class OrderItem {
  id!: number;
  quantity!: number;
  price!: number;
  orderId!: number;
  product!: Product;
  finalPrice!: number;
  productColorId!: number;
  productColor!: ProductColor;

  constructor(data?: any) {
    if (!data) return;

    this.id = data.id;
    this.quantity = data.quantity;
    this.price = data.price;
    this.orderId = data.orderId;
    this.product = data.product ? new Product(data.product) : undefined as any;
    this.finalPrice = data.finalPrice;
    this.productColorId = data.productColorId;
    this.productColor = data.productColor ? new ProductColor(data.productColor) : undefined as any;
  }
}
